### PR TITLE
[DOXIASITETOOLS-243] Drop Maven2 bits, update to Maven 3.2.5

### PR DESCRIPTION
**Pull request by cstamas**
Minimal changes to update to Maven 3.2.5 and get rid of Maven 2.0 bits.

---

https://issues.apache.org/jira/projects/DOXIASITETOOLS/issues/DOXIASITETOOLS-243

### CODE REVIEW

File: doxia-integration-tools/src/test/java/org/apache/maven/doxia/tools/SiteToolTest.java [hunk 1]
```diff
@@ -51,6 +53,12 @@
 public class SiteToolTest
     extends PlexusTestCase
 {
+    @Override
+    protected void customizeContainerConfiguration( @SuppressWarnings( "unused" ) final ContainerConfiguration configuration )
+    {
+        configuration.setClassPathScanning( PlexusConstants.SCANNING_CACHE );
```
Comment (michael-o) [inline doxia-integration-tools/src/test/java/org/apache/maven/doxia/tools/SiteToolTest.java:59]: That was my mistake, I simply missed this one.

File: doxia-integration-tools/pom.xml [hunk 2]
```diff
@@ -131,6 +145,74 @@
       <artifactId>maven-plugin-testing-harness</artifactId>
       <version>1.1</version>
```
Comment (michael-o) [inline doxia-integration-tools/pom.xml:146]: Maybe go straight to 3.3.0?

**Review by slachiewicz — APPROVED** [review]

**Comment by slawekjaranowski** [comment]
Is still needed:
```
if ( isMaven3OrMore() ) ...
```
and corresponding code in: `org.apache.maven.doxia.tools.DefaultSiteTool#getParentProject`

File: doxia-integration-tools/pom.xml (hunk 2 above)
Comment (cstamas) [reply doxia-integration-tools/pom.xml:146]: I think you are mixing up -harness and -tools?

Comment (michael-o) [reply doxia-integration-tools/pom.xml:146]: Yes, I think you are right.

**Comment by michael-o** [comment]
> Is still needed:
> 
> ```
> if ( isMaven3OrMore() ) ...
> ```
> 
> and corresponding code in: `org.apache.maven.doxia.tools.DefaultSiteTool#getParentProject`

Good catch, this can go.

File: doxia-integration-tools/pom.xml (hunk 2 above)
Comment (cstamas) [reply doxia-integration-tools/pom.xml]: fixed

File: doxia-integration-tools/pom.xml [hunk 3]
```diff
@@ -56,28 +56,37 @@
...
+      <artifactId>maven-core</artifactId>
       <version>${mavenVersion}</version>
       <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
-      <artifactId>maven-model</artifactId>
+      <artifactId>maven-compat</artifactId>
+      <version>${mavenVersion}</version>
+      <scope>provided</scope>
```
Comment (gnodet) [inline doxia-integration-tools/pom.xml:71]: Can't we drop this dependency to maven-compat ?

Comment (cstamas) [reply doxia-integration-tools/pom.xml:71]: Yup, should be simple to do.

**Comment by michael-o** [comment]
@cstamas Do you want to work on the compat removal or shall we merge this and handle compat separately?

**Pull request closed without merging**